packages/datetime/src/common/dateUtils.ts

```diff
@@ -309,3 +310,20 @@ export function stringToMoment(dateString: string, format: DateFormat, locale: s
         return date === undefined ? moment.invalid() : moment(date);
     }
 }
+
+
+export function getLocale(localeString: String): any {
+    if(!localeString) {
+        return undefined;
+    }
+    return require(`date-fns/locale/${localeString}`);
+}
+
+
+export function dateToString(date: Date, format: DateFormat, locale: string | undefined) {
+    if (typeof format === "string") {
+        return dateFnFormat(date, format, {locale: getLocale(locale)});
```

Comment:
it seems like if we dropped support for format strings then users could easily plug in any library like `{ dateToString: (date) => dateFnFormat(date, ...) }`.

this approach might solve the dynamic locale requires too?